src/vcpkg/versions.cpp

```diff
@@ -409,4 +409,98 @@ namespace vcpkg
 
         return static_cast<VerComp>(Util::range_lexcomp(a.identifiers, b.identifiers, uint64_comp));
     }
+
+    StringView normalize_external_version_zeros(StringView sv)
+    {
+        if (sv.empty()) return "0";
+
+        auto it = std::find_if_not(sv.begin(), sv.end(), [](char ch) { return ch == '0'; });
+        if (it == sv.end())
+        {
+            // all zeroes - just return "0"
+            return StringView{sv.end() - 1, sv.end()};
+        }
+        else
+        {
+            return StringView{it, sv.end()};
+        }
+    }
+
+    void ParsedExternalVersion::normalize()
+    {
+        major = normalize_external_version_zeros(major);
+        minor = normalize_external_version_zeros(minor);
+        patch = normalize_external_version_zeros(patch);
+    }
+
+    // /(\d\d\d\d)-(\d\d)-(\d\d).*/
+    bool try_parse_external_date_version(ParsedExternalVersion& out, StringView version)
```

Comment:
This one seems substantially identical to DateVersion's existing parser to me, should we reuse that one?